test/language/block-scope/syntax/for-in/acquire-properties-from-object.js

```diff
@@ -0,0 +1,21 @@
+// Copyright (C) Copyright 2011 the V8 project authors. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+/*---
+es6id: 13.1
+description: >
+    for-in to acquire properties from object
+includes: [compareArray.js]
+---*/
+function props(x) {
```

Comment:
You can get away without `Array.prototype.sort` and `compareArray.js` if you use an object literal:

```diff
 function props(x) {
-  var array = [];
+  var obj = {};
+  var count = 0;
-  for (let p in x) array.push(p);
+  for (let p in x) {
+    obj[p] = true;
+    count++;
+  }
-  return array.sort();
+  return { obj: obj, count: count }
 }
```

...but it's hard to tell if that's really worth it. Sometimes this feels more like code golf, so I won't mind if you tell me it is :)